giotto/homology/point_clouds.py

```diff
@@ -76,21 +81,23 @@ class VietorisRipsPersistence(BaseEstimator, TransformerMixin):
     """
     _hyperparameters = {'max_edge_length': [numbers.Number],
                         'infinity_values_': [numbers.Number],
-                        '_homology_dimensions': [list, [int, (0, np.inf)]]}
+                        '_homology_dimensions': [list, [int, (0, np.inf)]],
+                        'coeff': [int, (2, np.inf)]}
 
     def __init__(self, metric='euclidean', max_edge_length=np.inf,
-                 homology_dimensions=(0, 1), infinity_values=None,
+                 homology_dimensions=(0, 1), coeff=2, infinity_values=None,
                  n_jobs=None):
         self.metric = metric
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
         self.max_edge_length = max_edge_length
         self.infinity_values = infinity_values
-        self.homology_dimensions = homology_dimensions
         self.n_jobs = n_jobs
 
-    def _ripser_diagram(self, X, is_distance_matrix, metric):
-        Xds = ripser(X[X[:, 0] != np.inf], distance_matrix=is_distance_matrix,
-                     metric=metric, maxdim=self._max_homology_dimension,
-                     thresh=self.max_edge_length)['dgms']
+    def _ripser_diagram(self, X, is_distance_matrix, metric, coeff):
```

Comment:
I just notice that metric (and coeff) do not need to be passed to this class method as the are class parameters and can be accessed with self.